src/process_utility.c

```diff
@@ -872,6 +872,22 @@ process_altertable_drop_not_null(Hypertable *ht, AlterTableCmd *cmd)
 	}
 }
 
+static void
+process_altertable_drop_column(Hypertable *ht, AlterTableCmd *cmd)
+{
+	int			i;
+
+	for (i = 0; i < ht->space->num_dimensions; i++)
+	{
+		Dimension  *dim = &ht->space->dimensions[i];
+
+		if (strncmp(NameStr(dim->fd.column_name), cmd->name, NAMEDATALEN) == 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_IO_OPERATION_NOT_SUPPORTED),
+					 errmsg("cannot drop a partitioning column")));
```

Comment:
Let it be known in the error that this concerns a hypertable. Again, PG has these errors:

```
"cannot drop column named in partition key"
"cannot drop column referenced in partition key expression"
```

We should reuse those if possible but can add an `errdetail` specifying that this is a hypertab.e

Reply:
Is the second line the hint in postgres or sth else?